internal/authentication/ldap_user_provider.go

```diff
@@ -75,12 +75,12 @@ func (p *LDAPUserProvider) CheckUserPassword(username string, password string) (
 	}
 	defer adminClient.Close()
 
-	userDN, err := p.getUserDN(adminClient, username)
+	profile, err := p.getUserProfile(adminClient, username)
```

Comment:
getUserProfile actually doesn't use the ldap connection param you're passing to it "adminClient". It reconnects with a new connection, I doubt this is intentional.

Reply:
good catch!

Reply:
Should be fixed now.